packages/bpk-component-breakpoint/src/useMediaQuery.ts

```diff
@@ -16,15 +16,17 @@
  * limitations under the License.
  */
 
-import { useState, useEffect } from 'react';
+import { useEffect, useState } from 'react';
+
+const useMediaQuery = (query: string, matchSSR = false): boolean => {
+  const isClient = typeof window !== 'undefined' && !!window.matchMedia;
 
-const useMediaQuery = (query: string): boolean => {
   const [matches, setMatches] = useState(
-    window.matchMedia ? window.matchMedia(query).matches : false,
+    isClient ? window.matchMedia(query).matches : matchSSR,
   );
 
   useEffect(() => {
-    if (window.matchMedia) {
+    if (isClient) {
```

Comment:
the `useEffect` only runs on client, do we need this check? Unless it's possible that the `window` is defined, but `matchMedia` isn't? 🤔

Reply:
That is true, however, there are unit tests in backpack that are not mocking BpkBreakpoint, and so the useEffect runs and fails.

Reply:
why do they fail? I'm not sure I understand

Reply:
You can see from [this previous run](https://github.com/Skyscanner/backpack/actions/runs/8298787263/job/22713096089) that unit tests for components that relied on BpkBreakpoint were failing because:

- Unit tests do execute the `useEffect` unlike when SSR'ing
- Unit tests do not provide the full window object

Reply:
Ahh the GH actions logs are a mess 😅 I can only see
```
TypeError: window.matchMedia is not a function

      2 |
      3 | const useMediaQuery = (query: string) => {
    > 4 |   const [matches, setMatches] = useState(window.matchMedia(query).matches);
```
I would think that mocking the `matchMedia` on the window object would solve the issue?
Doesn't feel right to add this extra check just for unit tests to pass 🤔

Reply:
Added a mock to all of the tests that were failing due to this and removed the check within the useEffect

Reply:
I think we'll have to give some guidance to people, as any consumers of BpkBreakpoint, or any component that uses BpkBreakpoint will have to do mocks in their tests too.